src/Evolution/DiscontinuousGalerkin/Limiters/SimpleWenoImpl.hpp

```diff
@@ -0,0 +1,124 @@
+#include "Domain/Element.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/Mesh.hpp"
+#include "Evolution/DiscontinuousGalerkin/Limiters/WenoGridHelpers.hpp"
+#include "Evolution/DiscontinuousGalerkin/Limiters/WenoHelpers.hpp"
+#include "Evolution/DiscontinuousGalerkin/Limiters/WenoOscillationIndicator.hpp"
+#include "NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp"
+#include "NumericalAlgorithms/LinearOperators/MeanValue.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Limiters {
+namespace Weno_detail {
+
+// Compute the Simple WENO solution for one tensor component
+//
+// This interface is intended for use in limiters that check the troubled-cell
+// indicator independently for each tensor component. These limiters generally
+// need to limit only a subset of the tensor components.
+template <typename Tag, size_t VolumeDim, typename PackagedData>
+void simple_weno_impl(
+    const gsl::not_null<std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
+        intrp::RegularGrid<VolumeDim>,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>*>
+        interpolator_buffer,
+    const gsl::not_null<std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, DataVector,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>*>
+        modified_neighbor_solution_buffer,
+    const gsl::not_null<db::item_type<Tag>*> tensor,
+    const double neighbor_linear_weight, const size_t tensor_storage_index,
+    const Mesh<VolumeDim>& mesh, const Element<VolumeDim>& element,
+    const std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+        neighbor_data) noexcept {
+  // Compute the modified neighbor solutions.
+  // First extrapolate neighbor data onto local grid points, then shift the
+  // extrapolated data so its mean matches the local mean.
+  DataVector& component_to_limit = (*tensor)[tensor_storage_index];
+  const double local_mean = mean_value(component_to_limit, mesh);
+  for (const auto& neighbor_and_data : neighbor_data) {
+    const auto& neighbor = neighbor_and_data.first;
+    const auto& data = neighbor_and_data.second;
+
+    if (interpolator_buffer->find(neighbor) == interpolator_buffer->end()) {
+      // No interpolator found => create one
+      const auto& direction = neighbor.first;
+      const auto& source_mesh = data.mesh;
+      const auto target_1d_logical_coords =
+          Weno_detail::local_grid_points_in_neighbor_logical_coords(
+              mesh, source_mesh, element, direction);
+      (*interpolator_buffer)[neighbor] = intrp::RegularGrid<VolumeDim>(
+          source_mesh, mesh, target_1d_logical_coords);
+    }
+
+    // Avoid allocations by working directly in the preallocated buffer
+    DataVector& buffer = (*modified_neighbor_solution_buffer)[neighbor];
```

Comment:
I actually was creating the entries here... but I've changed this so that now I'm using `at` and preparing the buffer before calling `simple_weno_impl`.